### PR TITLE
ASTT-99 Added hovering functionality on buttons and drop downs

### PR DESCRIPTION
**Pull request by USilere**
This MR was to add more description to the buttons of the GUI, @shusi98  can add more of the styling, but find screenshots below to show work done on the PR.

Screenshot from 2024-05-09 12-00-48
Screenshot from 2024-05-09 11-58-46
Screenshot from 2024-05-09 11-38-22

### CODE REVIEW

File: src/astt_gui/templates/index.html [hunk 1]
```diff
@@ -202,15 +202,15 @@ <h1>CAM Dashboard</h1>
...
                     <div class="col">
                       <h4>Modes state</h4>
-                      <button class="btn btn-secondary" style="width: 190px;" id="modeState">UNKNOWN</button>
+                      <button class="btn btn-secondary" title="This button displays our modes state" style="width: 190px;" id="modeState">UNKNOWN</button>
                     </div>
                     <div class="col">
                       <h4>Functional State</h4>
-                      <button class="btn btn-secondary" style="width: 190px;" id="funcState">UNKNOWN</button>
+                      <button class="btn btn-secondary" title="This button displays our functional state" style="width: 190px;" id="funcState">UNKNOWN</button>
                     </div>
```
Comment (ssithole1) [inline src/astt_gui/templates/index.html:214]: These are buttons yes on the code but on the UX are displayed a grids so stating that  for e.g"This button displays our modes state" the user might expect that to be clickable for action, so I think we should reframe adding "button" on the hint for the 3 states

Comment (Silere8) [reply src/astt_gui/templates/index.html:214]: Done!

File: src/astt_gui/templates/index.html
Comment (USilere) [inline src/astt_gui/templates/index.html:1]: It seems imposible to center these @ssithole1 , I spoke to Zach and we tried the approach of using tooltips in CSS but the issue is how to the target the tooltip.. any suggestion??

Comment (USilere) [reply src/astt_gui/templates/index.html:1]: I also saw that the description moves as you move your mouse around the button so centering it seems futile. If I may, just keep the code as is and when you can you will create a ticket and style it.

File: src/astt_gui/templates/index.html [hunk 2]
```diff
@@ -226,26 +267,31 @@ <h4>Functional State</h4>
                         <div class="container">
                           <div class="row">
                             <div class="col">
-                              <button onclick="openInitModal('')" class="btn btn-primary"><i class='bx bxs-right-arrow'></i> Initialize System</button>
+                              <button onclick="openInitModal('')" class="btn btn-primary" title="This button initializes the system"> <i class='bx bxs-right-arrow'></i> Initialize System</button>
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:270]: This looks good and I tested it locally however, we can extend the statements to give more information about what the button does.

eg . Clicking this button will request for the machine's password.
	if the password is correct: 
	Then the antenna/simulator will be started and subscriptions to sensors is made on the antenna.
	then the initial values of sensors will be populated above on the GUI

Comment (USilere) [reply src/astt_gui/templates/index.html:270]: Will do.

File: src/astt_gui/templates/index.html [hunk 3]
```diff
@@ -201,22 +229,35 @@ <h1>CAM Dashboard</h1>
...
+                      </div> -->
+                      </div>
                     <div class="col">
                       <h4>Modes state</h4>
-                      <button class="btn btn-secondary" style="width: 190px;" id="modeState">UNKNOWN</button>
+                      <button class="btn btn-secondary" title="This is where we display our modes states" style="width: 190px;" id="modeState">UNKNOWN</button>
+                      <!-- <div class="btn-container">
+                        <button class="btn btn-secondary" id="modeState">UNKNOWN</button>
+                        <span class="tooltip">This is where we display our modes states</span>
+                      </div> -->
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:245]: Oops , Dead code.
Needs some cleaning.

File: src/astt_gui/templates/index.html [hunk 4]
```diff
@@ -201,22 +229,35 @@ <h1>CAM Dashboard</h1>
                 <div class="container">
                   <div class="row">
                     <div class="col">
-                     <h4>Stow-Pin State</h4>
-                     <button class="btn btn-secondary" style="width: 190px;" id="stowState">UNKNOWN</button>
-                    </div>
+                      <h4>Stow-Pin State</h4>
+                      <button class="btn btn-secondary" title="This is where we display our stow states" style="width: 190px;" id="stowState">UNKNOWN</button>
+                      <!--<div class="btn-container">
+                        <button class="btn btn-secondary" id="stowState">UNKNOWN</button>
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:235]: Oops , Dead code.
Needs some cleaning.

File: src/astt_gui/templates/index.html [hunk 5]
```diff
@@ -226,26 +267,31 @@ <h4>Functional State</h4>
                         <div class="container">
                           <div class="row">
                             <div class="col">
-                              <button onclick="openInitModal('')" class="btn btn-primary"><i class='bx bxs-right-arrow'></i> Initialize System</button>
+                              <button onclick="openInitModal('')" class="btn btn-primary" title="This button initializes the system"> <i class='bx bxs-right-arrow'></i> Initialize System</button>
                             </div>
                             <div class="col">
-                              <button class="btn btn-primary"><span class="bi bi-qr-code"></span> Scan Source/s</button>
+                              <button class="btn btn-primary" title="This button is used when we want to scan for sources i.e the sun also the stars in the evening"><span class="bi bi-qr-code"></span> Scan Source/s</button>
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:273]: This looks good and I tested it locally however, we can extend the statements to give more information about what the button does.

File: src/astt_gui/templates/index.html [hunk 6]
```diff
@@ -226,26 +267,31 @@ <h4>Functional State</h4>
...
+                              <!--<div class="btn-container">
+                                <button class="btn btn-primary"><span class="bi bi-qr-code"></span> Scan Source/s</button>
+                                <span class="tooltip">This is where we display sources</span>
+                              </div> -->
                             </div>
                             <div class="col">
-                              <button class="btn btn-primary"  data-bs-toggle="modal" data-bs-target="#modeModal"><span class='bi bi-arrows-collapse'></span> Antenna Modes</button>
+                              <button class="btn btn-primary" title="This button is used to switch between different antenna modes namely, Idle, Stow and Point."  data-bs-toggle="modal" data-bs-target="#modeModal"><span class='bi bi-arrows-collapse'></span> Antenna Modes</button>
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:280]: This looks good and I tested it locally however, we can extend the statements to give more information about what the button does.

File: src/astt_gui/templates/index.html [hunk 7]
```diff
@@ -226,26 +267,31 @@ <h4>Functional State</h4>
...
                             </div>
                             <div class="col">
-                              <button class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#pointModal"><span class="bi bi-app-indicator"></span> Point Source</button>
+                              <button class="btn btn-primary" title="This button allows us to point to a specific source, we give it pointing targets i.e azimuth and elevation in degrees" data-bs-toggle="modal" data-bs-target="#pointModal"><span class="bi bi-app-indicator"></span> Point Source</button>
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:283]: This looks good and I tested it locally however, we can extend the statements to give more information about what the button does.

File: src/astt_gui/templates/index.html [hunk 8]
```diff
@@ -226,26 +267,31 @@ <h4>Functional State</h4>
...
                             </div>
                             <div class="col">
-                              <button class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#trackModal"><span class='bi bi-geo-alt-fill'></span> Track Source</button>
+                              <button class="btn btn-primary" title="This button is used for tracking sources, the button allows you to choose a specific source you would like to track at a particular time." data-bs-toggle="modal" data-bs-target="#trackModal"><span class='bi bi-geo-alt-fill'></span> Track Source</button>
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:286]: This looks good and I tested it locally however, we can extend the statements to give more information about what the button does.

File: src/astt_gui/templates/index.html [hunk 9]
```diff
@@ -86,7 +114,7 @@
         <li class="nav-item dropdown">
 
           <a class="nav-link nav-icon" href="#" data-bs-toggle="dropdown">
-            <i class="bi bi-bell"></i><span>Alarms</span><span class="badge bg-danger badge-number">4</span>
+            <i class="bi bi-bell" title="This is an alert button for all warnings, errors and fatal system errors"></i><span>Alarms</span><span class="badge bg-danger badge-number">4</span>
```
Comment (AbednigoLethole) [inline src/astt_gui/templates/index.html:117]: This looks good and I tested it locally however, we can extend the statements to give more information about what the button does.

File: src/astt_gui/templates/index.html (hunk 4 above)
Comment (USilere) [reply src/astt_gui/templates/index.html:235]: This is what I was trying when I wanted to style the description but consulting Zach, he explained that it is not possible but I wanted surprise to also try from his side, if you can see I have new CSS file and a JS file check files changed.

**Review by AbednigoLethole — APPROVED** [review]

**Review by PhelokaziDube — APPROVED** [review]

**Pull request merged**